libs/wingc/src/type_check.rs

```diff
@@ -450,12 +450,60 @@ impl Subtype for Type {
 					parent_type.is_subtype_of(other)
 				})
 			}
-			(Self::Resource(res), Self::Interface(_)) => {
+			(Self::Resource(res), Self::Interface(iface)) => {
 				// If a resource implements the interface then it's a subtype of it (nominal typing)
-				res.implements.iter().any(|parent| {
+				let implements_iface = res.implements.iter().any(|parent| {
 					let parent_type: &Type = parent;
 					parent_type.is_subtype_of(other)
-				})
+				});
+
+				if implements_iface {
+					return true;
+				}
+
+				// To support flexible inflight closures, we say that any
+				// resource with an inflight method named "handle" is a subtype of
+				// any single-method interface with a matching "handle" method type.
+
+				// First, check if there is exactly one inflight method in the interface
```

Comment:
Note this might be clearer using the `ClassLike` trait implemented for interface. I think you can do something like:
```rust
  let inflight_methods = iface.methods().filter( /* is inflight and is called "handler" */);
  let handler_method = inflight_methods.next();
  if handler_method.is_none() || inflight_methods.next().is_some() { return false; }
```

Reply:
Nice, this worked 👍